src/extension.mligo

```diff
@@ -0,0 +1,94 @@
+#import "ligo-generic-fa2/lib/multi_asset/fa2.mligo" "FA2"
+#import "./errors.mligo" "Errors"
+
+type seconds = nat
+type permit_key = (address * bytes)
+type permits = (permit_key, timestamp) big_map
+type user_expiries = (address, seconds option) big_map
+type permit_expiries = (permit_key, seconds option) big_map
+
+type t = {
+    admin: address;
+    counter: nat;
+    default_expiry: seconds;
+    max_expiry: seconds;
+    permits: permits;
+    user_expiries: user_expiries;
+    permit_expiries: permit_expiries;
+}
+
+let get_expiry (ext: t) ((user, param_hash): permit_key) : seconds =
+    match Big_map.find_opt (user, param_hash) ext.permit_expiries with
+    | None ->
+        begin
+            match Big_map.find_opt user ext.user_expiries with
+            | None -> ext.default_expiry
+            | Some exp ->
+                begin
+                    match exp with
+                    | None -> ext.default_expiry
+                    | Some t -> t
+                end
+        end
+    | Some p ->
+        begin
+            match p with
+            | None -> ext.default_expiry
+            | Some exp -> exp
+        end
```

Comment:
```suggestion
    | Some (None) -> ext.default_expiry
    | Some (Some exp) -> exp
```
You can used nested/deep pattern matching here. But I'm not sure about the understandability of the code here.

Make this change only If you feel this will help in improving readabilty/understandability

Reply:
indeed, it is more readable, I see no problem as it is documented on ligolang